File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.rs

use std::collections::*;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32
    {
        let hash: HashSet<_> = nums.iter().cloned().collect();
        let mut max = 0;

        for mut num in nums
        {
            if hash.contains(&(num-1)) {
                continue;
            }

            let mut count = 0;

            while hash.contains(&num) {
                num += 1;
                count += 1;
            }
            max = max.max(count);
        }

        return max;
    }
}

File: Data Structures & Algorithms/number-of-one-bits/submission-0.cpp

class Solution {
public:
    int hammingWeight(uint32_t n)
    {
        int count = 0;

        for (uint32_t mask = 0x1; mask > 0; mask *= 2) {
            count += ((n & mask) == 0) ? 0 : 1;
        }
        return count;
    }
};
